yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -1337,55 +1337,8 @@ object Client extends Logging {
       conf: Configuration,
       credentials: Credentials) {
     if (shouldGetTokens(sparkConf, "hive") && UserGroupInformation.isSecurityEnabled) {
```

Comment:
As @tgravescs points out, the tokens are needed throughout the cluster, and yes, must be obtained irrespective of deployment.